### PR TITLE
F serf cmd

### PR DESCRIPTION
**Pull request by sean-**
Move the `serf` binary from the root of `github.com/hashicorp/serf` to `github.com/hashicorp/serf/cmd/serf`.  Serf's primary role is that of a library and therefore it should Serf should be as easy to embed out of the box.  The `make` and `make dev` targets remain unchanged and function as they did previously.

Fixes: #386 

Unrelated but worth noting that during the testing of this PR `darwin/arm` appears to be broken, but that looks like it's due to me using `go1.7rc6` and like this platform is now excluded from the build.

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -81,12 +81,6 @@
 			"revisionTime": "2015-12-06T04:24:12Z"
 		},
 		{
-			"checksumSHA1": "pWIsEUE53NffjS1goRml43Cvleg=",
```
Comment (slackpad) [inline vendor/vendor.json]: What's up with this diff?

Comment (sean-) [inline vendor/vendor.json]: `serf` now lo longer depends on `memberlist` so that it can be used as a library.

Comment (slackpad) [inline vendor/vendor.json]: Oh I was expecting that we'd have only a vendor down there under cmd - is that not the case?

**Comment by sean-** [comment]
OOB :shipit: from @slackpad

File: vendor/vendor.json (hunk 1 above)
Comment (sean-) [inline vendor/vendor.json]: Aye... "fixed."  See #404 and the updated comment I made at the bottom of #386 .

**Pull request merged**